include/message.h

```diff
@@ -0,0 +1,29 @@
+#ifndef __MESSAGE__
+#define __MESAAGE
+
+#include <string.h>
+
+#include "list.h"
+
+typedef enum {
+    PUSH_OK, PUSH_ERROR, PUSH_BLOCK
+} push_result;
+
+typedef struct group {
+    const char *name;
+    struct list_head same;
+    struct list_head unique;
+    void *state;
+    unsigned int data_size;
+    push_result (*push) (struct group *, void *, unsigned int);
+} channel_group;
+
+
+void init_message();
```

Comment:
no message anywhere, did you mean channel?
```suggestion
void channel_init();
```